server/aws-lsp-codewhisperer/src/language-server/inline-completion/session/sessionManager.ts

```diff
@@ -287,17 +288,17 @@ export class SessionManager {
     /**
      * Singleton SessionManager class
      */
-    public static getInstance(): SessionManager {
-        if (!SessionManager._instance) {
-            SessionManager._instance = new SessionManager()
+    public static getInstance(type: 'COMPLETIONS' | 'EDITS' = 'COMPLETIONS'): SessionManager {
+        if (type === 'EDITS') {
+            return (SessionManager._editInstance ??= new SessionManager())
         }
 
-        return SessionManager._instance
+        return (SessionManager._completionInstance ??= new SessionManager())
     }
 
     // For unit tests
     public static reset() {
-        SessionManager._instance = undefined
```

Comment:
this should reset both of its instances _completionInstance, _editInstance

Reply:
good callout, i miss this

Reply:
https://github.com/aws/language-servers/pull/2058/commits/880bf07e5a92019f4f5a613e28a985b73797486b